pkgs/tools/filesystems/nilfs-utils/default.nix

```diff
@@ -35,7 +35,16 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  configureFlags = [ "--with-libmount" ];
+  configureFlags = [
+    "--with-libmount"
+    # when cross-compiling we assume that the host system will return a valid
+    # pointer for calls to malloc(0) or realloc(0)
+    # https://www.uclibc.org/FAQ.html#gnu_malloc
+    # https://www.gnu.org/software/autoconf/manual/autoconf.html#index-AC_005fFUNC_005fMALLOC-454
+    # the upstream source should be patched to avoid needing this
+    "ac_cv_func_malloc_0_nonnull=yes"
```

Comment:
Since this is a hack, we should make it conditional on cross compiling. Otherwise we might be overriding legitimate detection.

Reply:
I wonder if it was possible to provide the correct information about the host in a more generic way to autotools, since I have seen this problem with `ac_cv_func_malloc_0_nonnull` in many places.